lib_comfyui/comfyui_reverse_proxy.py

```diff
@@ -0,0 +1,66 @@
+import os
+import sys
+import shutil
+import subprocess
+import threading
+import time
+import socket
+import importlib
+from lib_comfyui import argv_conversion, webui_settings
+importlib.reload(argv_conversion)
+importlib.reload(webui_settings)
+
+
+extension_root = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+process = None
+
+
+def start():
+    comfyui_argv = list(sys.argv)
+    argv_conversion.set_comfyui_argv(comfyui_argv)
+    port = comfyui_argv[comfyui_argv.index('--port') + 1]
+
+    npx_executable = shutil.which('npx')
+    if npx_executable is None:
+        return
+
+    threading.Thread(target=on_create_local_tunnel, args=(port, npx_executable, ), daemon=True).start()
+
+
+def on_create_local_tunnel(port, npx_executable):
+    global process
+    if process is not None:
+        return
+
+    wait_for_comfyui_started(int(port))
+
+    print("Launching localtunnel...")
+    process = subprocess.Popen(
+        [npx_executable, 'lt', "--port", port, "--local-host", '127.0.0.1'],
+        cwd=extension_root,
+        stdout=subprocess.PIPE)
```

Comment:
This process is not terminated when parent process dies. Gotta find a fix before merging.

Reply:
👍

Reply:
Popen.terminate() to terminate a subprocess instance:
"Popen.terminate()"
 https://docs.python.org/3/library/subprocess.html#:~:text=includes%20CREATE_NEW_PROCESS_GROUP.-,Popen.terminate(),-%C2%B6

Not sure when to call it though?